test-plans/src/lib.rs

```diff
@@ -0,0 +1,123 @@
+use std::{env, str::FromStr, time::Duration};
+
+use anyhow::{Context, Result};
+use env_logger::Env;
+use log::info;
+use redis::{AsyncCommands, Client as Rclient};
+use strum::EnumString;
+
+const REDIS_TIMEOUT: usize = 10;
+
+/// Supported transports by rust-libp2p.
+#[derive(Clone, Debug, EnumString)]
+#[strum(serialize_all = "kebab-case")]
+pub enum Transport {
+    Tcp,
+    QuicV1,
+    Webrtc,
+    Ws,
+}
+
+/// Supported stream multiplexers by rust-libp2p.
+#[derive(Clone, Debug, EnumString)]
+#[strum(serialize_all = "kebab-case")]
+pub enum Muxer {
+    Mplex,
+    Yamux,
+}
+
+/// Supported security protocols by rust-libp2p.
+#[derive(Clone, Debug, EnumString)]
+#[strum(serialize_all = "kebab-case")]
+pub enum SecProtocol {
+    Noise,
+    Tls,
+}
+
+/// Helper function to get a ENV variable into an test parameter like `Transport`.
+pub fn from_env<T>(env_var: &str) -> Result<T>
+where
+    T: FromStr,
+    T::Err: std::error::Error + Send + Sync + 'static,
+{
+    env::var(env_var)
+        .with_context(|| format!("{env_var} environment variable is not set"))?
+        .parse()
+        .map_err(Into::into)
+}
+
+/// PingSwarm allows us to abstract over libp2p versions for `run_ping`.
+#[async_trait::async_trait]
+pub trait PingSwarm: Sized + Send + 'static {
+    async fn listen_on(&mut self, address: &str) -> Result<String>;
+
+    fn dial(&mut self, address: &str) -> Result<()>;
+
+    async fn await_connections(&mut self, number: usize);
+
+    async fn await_pings(&mut self, number: usize) -> Vec<Duration>;
+
+    async fn loop_on_next(&mut self);
+
+    fn local_peer_id(&self) -> String;
+}
+
+/// Run a ping interop test. Based on `is_dialer`, either dial the address
+/// retrieved via `listenAddr` key over the redis connection. Or wait to be pinged and have
+/// `dialerDone` key ready on the redis connection.
+pub async fn run_ping<S>(
```

Comment:
Now that we only depend on one `libp2p` version, we can completely bin ( :wastebasket: ) this abstraction and have all the code in one place and properly reference types like `PeerId` etc!